testing/common/data/record/record_test.go

```diff
@@ -0,0 +1,763 @@
+package record
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"kantoku/common/data/record"
+	"kantoku/impl/common/data/record/dumb"
+	mongorec "kantoku/impl/common/data/record/mongo"
+	"log"
+	"math/rand"
+	"sort"
+	"testing"
+)
+
+func newMongoRecords(ctx context.Context) *mongorec.Storage {
+	// Set connection configurations
+	clientOptions := options.Client().ApplyURI("mongodb://localhost:27017")
+
+	// Connect to the MongoDB server
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+	go func() {
+		<-ctx.Done()
+
+		if err = client.Disconnect(ctx); err != nil {
+			log.Println("failed to disconnect from mongodb:", err)
+		}
+	}()
+
+	// Access the "main" database
+	database := client.Database("testing")
+
+	// Access the "testing" collection
+	collection := database.Collection("records")
+
+	return mongorec.New(collection)
+}
+
+var Keys = []string{
+	"updated_at",
+	"id",
+	"name",
+	"surname",
+	"group",
+}
+
+var Groups = []string{
+	"a",
+	"b",
+	"c",
+	"d",
+	"e",
+}
+
+const MinimumTestCases = 5
+const MaximumTestCases = 15
+const TotalRecords = 100
+
+func TestRecord(t *testing.T) {
+	rand.Seed(42)
+	ctx := context.Background()
+	implementations := map[string]record.Storage{
+		"mongo": newMongoRecords(ctx),
+	}
+
+	for label, _impl := range implementations {
+		impl := _impl
+
+		t.Run(label, func(t *testing.T) {
+			t.Log("Dataset initialization...")
+			t.Log("Erasing all data...")
+			if err := impl.Erase(ctx); err != nil {
+				t.Fatal("failed to erase:", err)
+			}
+
+			t.Log("Generating random records...")
+			specimen := dumb.New()
+			for i := 0; i < TotalRecords; i++ {
+				rec := randomRecord()
+				_ = specimen.Insert(ctx, rec)
```

Comment:
you sure you want to ignore error? It might be better to check if they both return error. But on the other hand, in this case they both shouldn't..

Reply:
1. no errors must occur during automated testing
2. specimen guarantees that no error happens during basic operations